### ISSUE TITLE
Docs: Restyle API methods, attributes and parameters into more readable format

### ISSUE THREAD

**Issue by marcysutton** [angular/material#984]
As part of https://github.com/angular/material/issues/916, API details should be presented in a data-table format. Whether this actually uses HTML tables or divs with ARIA roles is TBD ([related accessibility documentation](http://webaim.org/techniques/tables/data)).

The goal of this task is to make the API docs more readable by better utilizing visual space.

<p><strong>Existing mdBottomSheet documentation:</strong></p>
![Existing mdBottomSheet documentation](https://cloud.githubusercontent.com/assets/1045233/5493701/159a8318-86a2-11e4-9d86-8d17066c8588.png)

<p><strong>A better visual presentation for methods, taken from <a href="http://angular-ui.github.io/ui-router/site/#/api/ui.router.util.type:Type">UI Router</a>:</strong></p>
![Angular UI decode method](https://cloud.githubusercontent.com/assets/1045233/5493742/7239342a-86a2-11e4-9bc0-774d9198cc53.png)
